ethcore/src/miner/miner.rs

```diff
@@ -106,7 +106,8 @@ impl Miner {
 	#[cfg_attr(feature="dev", allow(cyclomatic_complexity))]
 	fn prepare_sealing(&self, chain: &MiningBlockChainClient) {
 		trace!(target: "miner", "prepare_sealing: entering");
-		let transactions = self.transaction_queue.lock().unwrap().top_transactions();
```

Comment:
This was actually working as expected - the lock should be dropped cause `top_transactions` is doing a copy.
I didn't wanted to keep transaction queue locked when transactions are executed.

But if you believe it might be dangerous I'm ok with it.